Parsers/CppParser/cppdocumentparser.cpp

```diff
@@ -184,8 +184,12 @@ void CppDocumentParser::reparseProject()
         return;
     }
     CppTools::CppModelManager *modelManager = CppTools::CppModelManager::instance();
-    QStringList list = d->activeProject->files(ProjectExplorer::Project::SourceFiles);
-    d->filesInStartupProject = list.filter(d->cppRegExp);
+    QStringList list;
+    foreach (Utils::FileName fileName, d->activeProject->files(ProjectExplorer::Project::SourceFiles)) {
+        if (fileName.toString().contains(d->cppRegExp))
+            list << fileName.toString();
+    }
+    d->filesInStartupProject = list;
```

Comment:
Can you use d->filesInStartupProject directly? There's no use for the "list" variable here.

Reply:
Impossible. It filters the list of filename.

Reply:
foreach (Utils::FileName fileName, d->activeProject->files(ProjectExplorer::Project::SourceFiles)) {
        if (fileName.toString().contains(d->cppRegExp))
            d->filesInStartupProject << fileName.toString();
}

Reply:
It is better. Thanks @becrux.